### PR TITLE
improve the behaviour of the engineering flot when reaching the limits of the double interval

### PR DESCRIPTION
**Pull request by cipix2000**
definitely not finished

### CODE REVIEW

File: dist/jquery.flot.js [hunk 1]
```diff
@@ -1535,10 +1570,6 @@ Licensed under the MIT license.
                                     val = +val; // convert to number
                                     if (isNaN(val)) {
                                         val = null;
-                                    } else if (val === Infinity) {
```
Comment (cipix2000) [inline dist/jquery.flot.js]: This needs to be tested. what happens if infinities reach the drawing functions ? clamt the values there ?

Maybe this part of the change dealinf with fakeInfinty should be rolled back

File: dist/jquery.flot.js [hunk 2]
```diff
@@ -2162,11 +2184,11 @@ Licensed under the MIT license.
                 ++dec;
             }
 
-            return dec;
+            return isFinite(dec) ? dec : 0;
```
Comment (ralucaportase) [inline dist/jquery.flot.js]: The precision could give us Infinity for example if our axis.max is greater than 1e308. In this case, make it 0.

File: dist/jquery.flot.js [hunk 3]
```diff
@@ -3066,10 +3070,14 @@ Licensed under the MIT license.
...
                     if (f === null || f === undefined) {
-                        continue
+                        continue;
                     }
 
-                    if ((!force && !f.computeRange) || val === fakeInfinity || val === -fakeInfinity) {
+                    if (typeof (isValid) === 'function' && !isValid(val)) {
+                        continue;
+                    }
+
```
Comment (ralucaportase) [inline dist/jquery.flot.js]: The validity of points should be checked for all points....

File: jquery.flot.logaxis.js [hunk 4]
```diff
@@ -98,6 +87,10 @@ Set axis.mode to "log" to enable.
             }
         });
 
+        if (maxIdx === -1) {
+            maxIdx = PREFERRED_LOG_TICK_VALUES.length - 1;
+        }
+
```
Comment (ralucaportase) [inline jquery.flot.logaxis.js]: For numbers greater than Number.MAX_VALUE, make maxIdx the one corresponding to MAX_VALUE

File: jquery.flot.js [hunk 5]
```diff
@@ -2405,11 +2400,15 @@ Licensed under the MIT license.
...
+                distance = series.datapoints.points[pointsize] - series.datapoints.points[0] || 1;
+
+            if (isFinite(distance)) {
+                minDistance = distance;
+            }
             for (var j = pointsize; j < series.datapoints.points.length - pointsize; j += pointsize) {
                 distance = Math.abs(series.datapoints.points[pointsize + j] - series.datapoints.points[j]);
-                if (distance < minDistance) {
+                if (distance < minDistance && isFinite(distance)) {
                     minDistance = distance;
```
Comment (ralucaportase) [inline jquery.flot.js]: The distance could not be finite if we have for example null - infinity

File: jquery.flot.absRelTime.js [hunk 6]
```diff
@@ -193,7 +199,16 @@ API.txt for details.
...
     function dateGenerator(ts, opts) {
+        var maxDateValue = 8640000000000000;
+
         ts *= 1000;
+
+        if (ts > maxDateValue) {
+            ts = maxDateValue;
+        } else if (ts < -maxDateValue) {
+            ts = -maxDateValue;
+        }
```
Comment (ralucaportase) [inline jquery.flot.absRelTime.js]: Date() works only for values in range (-8640000000000000; 8640000000000000)

**Pull request merged**